src/katsdpcalproc/pointing.py

```diff
@@ -1,87 +1,87 @@
-#! /usr/bin/env python
+##! /usr/bin/env python
 #
-# Perform a collection of offset pointings on the nearest pointing calibrator.
-# Obtain interferometric gain solutions from the pipeline.
-# Fit primary beams to the gains and calculate pointing offsets from them.
-# Store pointing offsets in telstate.
+# Updated refrence pointing script
+# This script contains low level functions to calculate
+# offsets of a reference pointing observation
+# The following low-level functions are used to calculate the offsets:
+# Extract gains per pointing offset, per receptor and per frequency chunk.
+# Fit primary beams to the gains
+# Calculate offsets from fitted beams
+# Ludwig Schwardt & Tasmiyah Rawoot
 #
-# Ludwig Schwardt
-# 2 May 2017
-#
-
-import time
 
+import katpoint
 import numpy as np
-from katcorelib.observe import (standard_script_options, verify_and_connect,
-                                collect_targets, start_session, user_logger,
-                                CalSolutionsUnavailable)
-from katpoint import (rad2deg, deg2rad, lightspeed, wrap_angle,
-                      RefractionCorrection)
+from katpoint import (rad2deg, deg2rad, lightspeed, wrap_angle, RefractionCorrection)
 from scikits.fitting import ScatterFit, GaussianFit
 
 
-# Group the frequency channels into this many sections to obtain pointing fits
-NUM_CHUNKS = 16
-
+# Test that NUM_CHUNKS is a multiple of number of frequency channels
+class NotMUltipleError(Exception):
+    pass
 
-class NoTargetsUpError(Exception):
-    """No targets are above the horizon at the start of the observation."""
 
-
-def get_offset_gains(session, offsets, offset_end_times, track_duration):
+def get_offset_gains(bp_gains, gains, offsets, NUM_CHUNKS, ants,
+                     track_duration, centre_freq, bandwidth, no_channels, pols):
     """Extract gains per pointing offset, per receptor and per frequency chunk.
 
     Parameters
     ----------
-    session : :class:`katcorelib.observe.CaptureSession` object
-        The active capture session
-    offsets : sequence of *N* pairs of float (i.e. shape (*N*, 2))
-        Requested (x, y) pointing offsets relative to target, in degrees
-    offset_end_times : sequence of *N* floats
-        Unix timestamp at the end of each pointing track
-    track_duration : float
-        Duration of each pointing track, in seconds
+    bp_gains : a list containing a numpy array of shape (no.channels,
+        no.polarizations,no.antennas) for each pointing offset *N*, ie.
+        for N pointing offsets, len(bp_gains)=N.
```

Comment:
How sure are you that there will be exactly `N` sets of gains? The previous approach allowed for failed gain solutions, not sure if still needed...